finetune_adapter.py

```diff
@@ -0,0 +1,230 @@
+import shutil
+import time
+from pathlib import Path
+
+import lightning as L
+import numpy as np
+import torch
+from lightning.fabric.strategies import DeepSpeedStrategy
+
+from generate import generate
+from lit_stablelm.adapter import StableLM, Config, mark_only_adapter_as_trainable, adapter_state_from_state_dict
+from lit_stablelm.tokenizer import Tokenizer
+from lit_stablelm.utils import EmptyInitOnDevice, lazy_load, check_valid_checkpoint_dir
+from scripts.prepare_alpaca import generate_prompt
+
+eval_interval = 600
+save_interval = 1000
+eval_iters = 100
+log_interval = 1
+devices = 1
+
+# Hyperparameters
+learning_rate = 9e-3
+batch_size = 64 / devices
+micro_batch_size = 1
+gradient_accumulation_steps = batch_size // micro_batch_size
+epoch_size = 50000  # train dataset size
+num_epochs = 5
+max_iters = num_epochs * epoch_size // devices
+weight_decay = 0.02
+max_seq_length = 256  # see scripts/prepare_alpaca.py
+warmup_steps = epoch_size * 2 // micro_batch_size // devices  # 2 epochs
+
+ds_config = {
+    "train_micro_batch_size_per_gpu": micro_batch_size,
+    "gradient_accumulation_steps": gradient_accumulation_steps,
+    "zero_optimization": {
+        "stage": 2,
+        # "overlap_comm": True,
+        # "allgather_partitions":True,
+        # "allgather_bucket_size": 2e8,
+        # "reduce_scatter":True,
+        # "contiguous_gradients": True,
+        # "round_robin_gradients": True,
+        "offload_optimizer": {"device": "cpu", "pin_memory": True},
+        "offload_param": {"device": "cpu"},
+    },
+}
+
+
+def main(
+    data_dir: Path = Path("data/alpaca"), checkpoint_dir: Path = Path("checkpoints/stabilityai/stablelm-base-alpha-3b")
+):
+    check_valid_checkpoint_dir(checkpoint_dir)
+
+    fabric = L.Fabric(
+        accelerator="cuda",
+        devices=devices,
+        strategy=(DeepSpeedStrategy(config=ds_config) if devices > 1 else "auto"),
+        precision="bf16-mixed",
```

Comment:
Lit-llama uses `bf16-true` here after https://github.com/Lightning-AI/lit-llama/pull/201, which landed before this PR if I'm not mistaken.

@ArturK-85 did you set `mixed` here on purpose? If not, we should standardize